Fix errors below when built with gcc-4.1.2

sysutil.c: In function 'vsf_sysutil_wait_exited_normally':
sysutil.c:604: error: assignment of read-only member '__in'
sysutil.c: In function 'vsf_sysutil_wait_get_exitcode':
sysutil.c:614: error: assignment of read-only member '__in'

--- vsftpd-2.0.4-old/sysutil.c
+++ vsftpd-2.0.4-new/sysutil.c
@@ -601,7 +601,7 @@
 vsf_sysutil_wait_exited_normally(
   const struct vsf_sysutil_wait_retval* p_waitret)
 {
-  return WIFEXITED(p_waitret->exit_status);
+  return WIFEXITED(((struct vsf_sysutil_wait_retval *)p_waitret)->exit_status);
 }
 
 int
@@ -611,7 +611,7 @@
   {
     bug("not a normal exit in vsf_sysutil_wait_get_exitcode");
   }
-  return WEXITSTATUS(p_waitret->exit_status);
+  return WEXITSTATUS(((struct vsf_sysutil_wait_retval *)p_waitret)->exit_status);
 }
 
 void
